polus-feature-extraction-plugin/src/utility.py

```diff
@@ -0,0 +1,1035 @@
+                'standard_deviation': standard_deviation,
+                'skewness': skewness,
+                'entropy': entropy,
+                'kurtosis': kurtosis,
+                'neighbors': neighbors,
+                'maxferet': maxferet,
+                'minferet': minferet,
+                'polygonality_score': polygonality_score,
+                'hexagonality_score': hexagonality_score,
+                'hexagonality_sd': hexagonality_sd,
+                'all': all}
+      
+        #Calculate features given as input for all images
+        regions = measure.regionprops(self.label_image,self.intensity_image)#region properties
+        title = self.filenames#to pass the filename in csv
+        
+        for each_feature in self.feature:
+            label = [r.label for r in regions]#get labels list for all regions
+            feature_value = FEAT[each_feature](self.label_image,self.intensity_image)#dynamically call the function based on the features required
+            
+            #get all features
+            if each_feature  == 'all':
+                df=pd.DataFrame(feature_value)#create dataframe for all features
+                df = df.T#transpose
+                df.columns =['Area','Centroid row','Centroid column','Convex area','Eccentricity','Equivalent diameter','Euler number','Hexagonality score','Hexagonality sd','Kurtosis','Major axis length','Maxferet','Maximum intensity','Mean intensity','Median','Minimum intensity','Minferet','Minor axis length','Mode','Neighbors','Orientation','Perimeter','Polygonality score','Standard deviation','Skewness','Solidity']
+            else:    
+                df = pd.DataFrame({each_feature: feature_value})#create dataframe for features selected
+            self.df_insert = pd.concat([self.df_insert, df], axis=1)    
+        self.df_insert.insert(0, 'Image', title)#Insert filename as 1st column 
+        self.df_insert.insert(1, 'Label', label)#Insert label as 2nd column
+        self.df_insert.columns = map (lambda x: x.capitalize(),self.df_insert.columns)#Capitalize the first letter of header
+        
+        #save each cav file separately
+        if self.csv_file == 'csvmany':
+            csv_file= Df_Csv_multiple(self.df_insert, self.output_dir,title)
+            csv_final = csv_file.csvfilesave()
+        else:
+            self.df_csv = self.df_csv.append(self.df_insert)
+        return self.df_csv
+
+#class to save features for all images in same file    
+class Df_Csv_single(Analysis):
+    def __init__(self,df_csv, output_dir):
+        self.df_export = df_csv
+        self.output_dir = output_dir
+ 
+    def csvfilesave(self):
+        os.chdir(self.output_dir)
+        #Export to csv
+        export_csv = self.df_export.to_csv (r'Feature_Extraction.csv', index = None, header=True)
+        del self.df_export
+        del export_csv
+
+#class to save features for each image in separate file          
+class Df_Csv_multiple(Analysis):
```

Comment:
Add class and method commenting according to the Google Python commenting specification.